state/state.go

```diff
@@ -2021,8 +2024,14 @@ func (st *State) AddRelation(eps ...Endpoint) (r *Relation, err error) {
 		if exists, err := isNotDead(st, relationsC, key); err != nil {
 			return nil, errors.Trace(err)
 		} else if exists {
+			// Ignore the error here, if there is an error, we know that dying
+			// will be false and can fall through to error message below.
+			if dying, _ := isDying(st, relationsC, key); dying {
+				return nil, errors.AlreadyExistsf("relation %v is dying and", key)
```

Comment:
I think something like:
```go
errors.NewAlreadyExists(nil, fmt.Sprintf("relation %v is dying, but not yet removed", key))
```
although it not as pleasing in code, results in a better error message:
- We indicate the state better without burdening the user with the specific knowledge of the `life` concept.
- We are not being tautological with "already exists".

Reply:
Always forget about generating the error without the suffix using New!